### PR TITLE
Rearchitect for multi-package publishing

### PR DESCRIPTION
**Pull request by nhuntwalker**
@Joll59 needs to add his email to the package.json files in the base repos as well as the sub-modules and do the lerna publish. The architecture is all there though.

### CODE REVIEW

File: src/cris-test-harness-cli/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "cris-test-harness-cli",
```
Comment (Joll59) [inline src/cris-test-harness-cli/package.json]: Naming consistency: d-ser-t-cli ?

File: src/cris-test-harness-cli/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "version": "0.3.0",
+  "description": "Package for running the cris-test-harness from the command line",
+  "main": "lib/main.js",
+  "types": "lib/main.d.ts",
+  "scripts": {
+    "build": "npx tsc"
+  },
+  "author": "@jolly59, @nhuntwalker",
```
Comment (Joll59) [inline src/cris-test-harness-cli/package.json]: @KatieProchilo , should be added as well.

File: src/cris-test-harness-cli/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "@types/async": "^2.4.2",
+    "@types/diff": "^4.0.2",
+    "@types/jest": "^24.0.13",
+    "@types/node": "^12.0.0",
+    "@types/yargs": "^13.0.0",
+    "jest": "^24.8.0",
+    "ts-jest": "^24.0.2",
+    "ts-node": "^8.1.0"
+  },
+  "gitHead": "731c25486adc924a180e04f9dad46ee8bf4eb6f2"
```
Comment (Joll59) [inline src/cris-test-harness-cli/package.json]: is this `githead` pointing at the current d-ser-t repo ?

File: src/cris-test-harness-cli/package.json (hunk 1 above)
Comment (nhuntwalker) [reply src/cris-test-harness-cli/package.json]: will do for both then

File: src/cris-test-harness-cli/package.json (hunk 2 above)
Comment (nhuntwalker) [reply src/cris-test-harness-cli/package.json]: Done

File: src/cris-test-harness-cli/package.json (hunk 3 above)
Comment (nhuntwalker) [reply src/cris-test-harness-cli/package.json]: The `githead` I'm seeing in `package.json` now is pointing at the `nmhw/lerna` branch in this repository.

**Comment by Joll59** [comment]
Rebased onto a different branch and PR'd: #44 .

**Pull request merged**